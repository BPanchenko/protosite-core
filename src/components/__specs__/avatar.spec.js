import transform from 'lodash/transform'
import AvatarComponent, { settings } from '../avatar.js'

describe('AvatarComponent', () => {
	it('Settings', () => {
		expect(settings).toMatchSnapshot()
	})

	it('Shadow DOM', () => {
		const avatar = new AvatarComponent()
		expect(avatar.shadowRoot.children).toMatchSnapshot()
	})

	describe.each`
		link                   | image
		${undefined}           | ${undefined}
		${'http://yandex.ru'}  | ${'http://protosite.rocks/pictures/ava.gif'}
		${undefined}           | ${'http://protosite.rocks/pictures/ava-1.gif'}
		${'http://google.com'} | ${undefined}
	`('Link = "$link"; Image = "$image"', (options) => {
		const dataset = transform(
			options,
			(result, value, key) =>
				value !== undefined && (result[key] = value),
			{},
		)
		const avatar = new AvatarComponent(dataset)

		beforeEach(() => document.body.appendChild(avatar))
		afterEach(() => avatar.remove())

		it('Light DOM', () => {
			expect(avatar).toMatchSnapshot()
		})
	})
})
