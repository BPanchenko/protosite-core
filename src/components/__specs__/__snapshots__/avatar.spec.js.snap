// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarComponent Custom Methods and Properties 1`] = `
Object {
  "connectedCallback": [Function connectedCallback],
}
`;

exports[`AvatarComponent Link = "http://google.com"; Image = "undefined" markup 1`] = `
<body>
  <template
    id="c-avatar_tpl1"
  />
  Object {}
</body>
`;

exports[`AvatarComponent Link = "http://yandex.ru"; Image = "http://protosite.rocks/pictures/ava.gif" markup 1`] = `
<body>
  <template
    id="c-avatar_tpl1"
  />
  Object {}
</body>
`;

exports[`AvatarComponent Link = "undefined"; Image = "http://protosite.rocks/pictures/ava-1.gif" markup 1`] = `
<body>
  <template
    id="c-avatar_tpl1"
  />
  Object {}
</body>
`;

exports[`AvatarComponent Link = "undefined"; Image = "undefined" markup 1`] = `
<body>
  <template
    id="c-avatar_tpl1"
  />
  Object {}
</body>
`;

exports[`AvatarComponent Settings 1`] = `
Object {
  "cssText": ":root{--clr-gray-100:#f3f3f3;--clr-gray-300:#c6c6c6;--clr-gray-400:#ababab}html{font-size:62.5%}:root{--muted-color:var(--clr-gray-400);--shadow-key-umbra-opacity:0.2;--shadow-key-penumbra-opacity:0.14;--shadow-ambient-shadow-opacity:0.12;--shadow:var(--shadow-4dp);--shadow-4dp:0 4px 5px 0 rgba(0,0,0,var(--shadow-key-penumbra-opacity)),0 1px 10px 0 rgba(0,0,0,var(--shadow-ambient-shadow-opacity)),0 2px 4px -1px rgba(0,0,0,var(--shadow-key-umbra-opacity))}.c-avatar{--bg-color:transparent;--border:max(1 * var(--pixel)) solid var(--muted-color);--shadow:none;--size:max(5 * var(--unit));background-color:var(--bg-color);border:var(--border);border-radius:50%;box-shadow:var(--shadow);display:inline-block;height:var(--size);margin:0;overflow:hidden;text-align:center;user-select:none;vertical-align:middle;width:var(--size)}.c-avatar :where(a,img){display:inline-block;height:inherit;width:inherit}.c-avatar img{object-fit:cover}:is(.c-avatar[style*=background-image]){background-size:cover}:is(.c-avatar:not([style*=background-image])):empty{--bg-color:var(--clr-gray-100);--border:none;--shadow:none}:is(.c-avatar:not([style*=background-image])) .c-avatar__link:empty{box-sizing:border-box;position:relative}:is(.c-avatar:not([style*=background-image])) .c-avatar__link:empty:before{background-color:var(--clr-gray-300);border-radius:50%;content:"";display:inline-block;height:40%;left:50%;position:absolute;top:50%;transform:translate(-50%,-50%);width:40%}",
  "tagName": "c-avatar",
  "template": "<figure class="c-avatar"></figure>",
}
`;
